lib/config/configuration.go

```diff
@@ -694,11 +694,13 @@ func applyProxyConfig(fc *FileConfig, cfg *service.Config) error {
 			return trace.BadParameter("kube_listen_addr must be set when kube_public_addr is set")
 		}
 		cfg.Proxy.Kube.Enabled = true
-		addr, err := utils.ParseHostPortAddr(fc.Proxy.KubeAddr, int(defaults.KubeListenPort))
-		if err != nil {
-			return trace.Wrap(err)
+		if fc.Proxy.KubeAddr != "" {
```

Comment:
my bad, this check is not needed because this is newKube switch case flow https://github.com/gravitational/teleport/blob/ab8ec4ae971aa2846baa266196fdfce8c86e39b3/lib/config/configuration.go#L666 thus we are sure that `fc.Proxy.KubeAddr` was set.